javaparser-core-testing/src/test/java/com/github/javaparser/ast/validator/Java5ValidatorTest.java

```diff
@@ -133,6 +133,15 @@ void foreach() {
         assertNoProblems(result);
     }
 
+    @Test
+    void noModifiersInForEachBesideFinal() {
+        ParseResult<Statement> result = javaParser.parse(STATEMENT, provider("for(static transient int i : nums){}"));
+        assertProblems(result,
+                "(line 1,col 5) 'static' is not allowed here.",
+                "(line 1,col 5) 'transient' is not allowed here.",
+                "(line 1,col 1) A foreach statement's variable declaration may have at most one 'final' modifier, and no other modifiers. Given: [static , transient ].");
```

Comment:
Hmmm, three messages are a little overkill?

Reply:
Sure, what should we do instead? The two first messages are already generated by existing validators; those are not my work. :-)